.pre-commit-config.yaml

```diff
@@ -4,22 +4,14 @@
 #   pre-commit autoupdate
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.3.0
+    rev: v4.4.0
     hooks:
       - id: trailing-whitespace
       - id: end-of-file-fixer
         exclude: '.*\.pth$'
       - id: debug-statements
-  - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.253
     hooks:
-      - id: isort
-  - repo: https://github.com/asottile/pyupgrade
-    rev: v3.2.2
-    hooks:
-      - id: pyupgrade
-        args: [--py36-plus]
-  - repo: https://github.com/PyCQA/flake8
-    rev: 5.0.4
-    hooks:
-      - id: flake8
+      - id: ruff
+        # args: [--fix, --exit-non-zero-on-fix]
```

Comment:
It would be convenient to have autofix, (pyupgrade was autofixing so it would be a regression not to)

Reply:
Can we add `--fix` in a later PR?  They tend to be more controversial and generate a lot of [redo requests](https://github.com/pytest-dev/pytest-cov/pull/583#issuecomment-1447794026).

Reply:
`--show-fixes` is necessary to show what has been fixed (and more importantly why).

I think enabling `--fix` in a later pr only delays the inevitable. This is like buying new alloy rims, taking the wheels off your car and throw the old rims with the old tires and then have it stuck on blocks forever cause you don't have the right tires for the new rims.